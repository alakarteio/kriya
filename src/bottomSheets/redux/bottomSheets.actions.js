export const OPEN_BOTTOMSHEETS = 'OPEN_BOTTOMSHEETS'
export const open = (code, contents) => ({ type: OPEN_BOTTOMSHEETS, payload: { code, contents } })

export const CLOSE_BOTTOMSHEETS = 'CLOSE_BOTTOMSHEETS'
export const close = code => ({ type: CLOSE_BOTTOMSHEETS, payload: code })

export const TOGGLE_BOTTOMSHEETS = 'TOGGLE_BOTTOMSHEETS'
export const toggle = code => ({ type: TOGGLE_BOTTOMSHEETS, payload: code })
