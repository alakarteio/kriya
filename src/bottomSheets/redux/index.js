import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
