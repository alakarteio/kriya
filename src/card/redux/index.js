export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
