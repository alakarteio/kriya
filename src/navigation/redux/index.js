export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
