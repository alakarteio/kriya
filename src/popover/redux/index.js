export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
