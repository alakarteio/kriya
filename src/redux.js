export { default as cards } from './card/redux'
export { default as navigation } from './navigation/redux'
export { default as popover } from './popover/redux'
export { default as bottomSheets } from './bottomSheets/redux'
export { default as tabs } from './tabs/redux'
