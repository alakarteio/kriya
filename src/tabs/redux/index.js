export { default } from './tabs'
export * from './tabs.actions'
export * from './tabs.selectors'
