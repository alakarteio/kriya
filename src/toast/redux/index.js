import toast from './toast'
import * as actions from './toast.actions'
import * as selectors from './toast.selectors'

Object.assign(toast, actions, selectors)

export * from './toast.actions'
export * from './toast.selectors'
export default toast

