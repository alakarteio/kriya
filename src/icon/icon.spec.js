/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import Icon from './icon'
import IconContainer from './icon.container'

describe('common/Icon', () => {
  describe('graphical (JSX)', () => {
    const component = ({ ...props }) => {
      return renderer.create(
        <Icon {...props} />,
      )
    }

    const snap = (props) => {
      const tree = component({ prefix: 'aprefix-', children: 'icon', ...props }).toJSON()
      expect(tree).toMatchSnapshot()
    }

    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should create a primary icon', () => snap({ primary: true }))
    it('should create a secondary icon', () => snap({ secondary: true }))
    it('should create a disabled icon', () => snap({ disabled: true }))
    it('should create an icon without color', () => snap({ noColor: true }))
    it('should create an icon with label to left', () => snap({ labelLeft: 'label' }))
    it('should create an icon with label to right', () => snap({ labelRight: 'label to right' }))
    it('should print labels to both sides', () => snap({ labelLeft: 'left', labelRight: 'right' }))
    it('should take an icon name as children', () => snap({ children: 'different-icon' }))
    it('should take a prefix', () => snap({ children: 'myIcon', prefix: 'myPrefix-' }))
    it('should have a default behaviour', () => snap({}))
  })

  describe('container', () => {
    /* eslint-disable react/no-children-prop */
    const snap = (state, { children = 'icon', ...props }) => {
      const store = createStore(() => ({ config: state }))
      const component = renderer.create(
        <Provider store={store}>
          <IconContainer children={children} {...props} />
        </Provider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    }
    /* eslint-enable react/no-children-prop */

    it('should looking for prefix in redux', () => snap(
      { icon: { prefix: 'redux-prefix' } },
      {},
    ))

    it('should use a default prefix (empty one)', () => snap(
      {},
      {},
    ))
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
