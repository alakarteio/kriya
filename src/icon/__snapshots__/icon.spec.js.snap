// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Icon container should looking for prefix in redux 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <i
    className="redux-prefixicon"
    name="redux-prefixicon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon container should use a default prefix (empty one) 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <i
    className="icon"
    name="icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should add custom className 1`] = `
<div
  className="icon custom colored"
  style={Object {}}
>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should add custom style 1`] = `
<div
  className="icon colored"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create a disabled icon 1`] = `
<div
  className="icon disabled colored"
  style={Object {}}
>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create a primary icon 1`] = `
<div
  className="icon primary colored"
  style={Object {}}
>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create a secondary icon 1`] = `
<div
  className="icon secondary colored"
  style={Object {}}
>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create an icon with label to left 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <label
    htmlFor="aprefix-icon"
  >
    label
  </label>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should create an icon with label to right 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
  <label
    htmlFor="aprefix-icon"
  >
    label to right
  </label>
</div>
`;

exports[`common/Icon graphical (JSX) should create an icon without color 1`] = `
<div
  className="icon"
  style={Object {}}
>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should have a default behaviour 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should print labels to both sides 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <label
    htmlFor="aprefix-icon"
  >
    left
  </label>
  <i
    className="aprefix-icon"
    name="aprefix-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
  <label
    htmlFor="aprefix-icon"
  >
    right
  </label>
</div>
`;

exports[`common/Icon graphical (JSX) should take a prefix 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <i
    className="myPrefix-myIcon"
    name="myPrefix-myIcon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;

exports[`common/Icon graphical (JSX) should take an icon name as children 1`] = `
<div
  className="icon colored"
  style={Object {}}
>
  <i
    className="aprefix-different-icon"
    name="aprefix-different-icon"
  >
    <span
      className="path1"
    />
    <span
      className="path2"
    />
    <span
      className="path3"
    />
    <span
      className="path4"
    />
    <span
      className="path5"
    />
    <span
      className="path6"
    />
    <span
      className="path7"
    />
    <span
      className="path8"
    />
    <span
      className="path9"
    />
    <span
      className="path10"
    />
    <span
      className="path11"
    />
    <span
      className="path12"
    />
    <span
      className="path13"
    />
  </i>
</div>
`;
